runtime/doc/lsp.txt

```diff
@@ -39,53 +41,96 @@ To check LSP clients attached to the current buffer:  >
-  nnoremap <silent> g0    <cmd>lua vim.lsp.buf.document_symbol()<CR>
-  nnoremap <silent> gW    <cmd>lua vim.lsp.buf.workspace_symbol()<CR>
-  nnoremap <silent> gd    <cmd>lua vim.lsp.buf.declaration()<CR>
+Example config (in init.vim): >
 
-Note: Language servers may have limited support for these features.
+  lua << EOF
+    local custom_lsp_attach = function(client)
+      -- See `:help nvim_buf_set_keymap()` for more information
+      vim.api.nvim_buf_set_keymap(0, 'n', 'K', '<cmd>lua vim.lsp.buf.hover()<CR>', {noremap = true})
+      vim.api.nvim_buf_set_keymap(0, 'n', '<c-]>', '<cmd>lua vim.lsp.buf.definition()<CR>', {noremap = true})
+      -- ... and other keymappings for LSP
 
-Nvim provides the |vim.lsp.omnifunc| 'omnifunc' handler which allows
-|i_CTRL-X_CTRL-O| to consume LSP completion. Example config (note the use of
-|v:lua| to call Lua from Vimscript): >
+      -- Use LSP as the handler for omnifunc.
+      --    See `:help omnifunc` and `:help ins-completion` for more information.
+      vim.api.nvim_buf_set_option(0, 'omnifunc', 'v:lua.vim.lsp.omnifunc')
 
-  " Use LSP omni-completion in Python files.
-  autocmd Filetype python setlocal omnifunc=v:lua.vim.lsp.omnifunc
+      -- For plugins with an `on_attach` callback, call them here. For example:
+      -- require('completion').on_attach(client)
+    end
 
-If a function has a `*_sync` variant, it's primarily intended for being run
-automatically on file save.  E.g. code formatting: >
+    -- An example of configuring for `sumneko_lua`,
+    --  a language server for Lua.
+    -- First, you must run `:LspInstall sumneko_lua` for this to work.
+    require('nvim_lsp').sumneko_lua.setup({
+      -- An example of settings for an LSP server.
+      --    For more options, see nvim-lspconfig
+      settings = {
+        Lua = {
+          diagnostics = {
+            enable = true,
+            globals = { "vim" },
+          },
+        }
+      },
+
+      on_attach = custom_lsp_attach
+    })
+  EOF
+<
 
-  " Auto-format *.rs files prior to saving them
-  autocmd BufWritePre *.rs lua vim.lsp.buf.formatting_sync(nil, 1000)
+Full list of features provided by default can be found in |lsp-buf|. 
 
 ================================================================================
 FAQ                                                     *lsp-faq*
 
 - Q: How to force-reload LSP?
   A: Stop all clients, then reload the buffer. >
 
-  :lua vim.lsp.stop_client(vim.lsp.get_active_clients())
-  :edit
+     :lua vim.lsp.stop_client(vim.lsp.get_active_clients())
+     :edit
 
 - Q: Why isn't completion working?
   A: In the buffer where you want to use LSP, check that 'omnifunc' is set to
-  "v:lua.vim.lsp.omnifunc": >
+     "v:lua.vim.lsp.omnifunc": >
+
+     :verbose set omnifunc?
+
+<     Some other plugin may be overriding the option. To avoid that, you could
+     set the option in an |after-directory| ftplugin, e.g.
+     "after/ftplugin/python.vim".
```

Comment:
Perhaps also include a way of checking if the LSP is running?

Reply:
We already mention above the trick of doing:

```
  :lua print(vim.inspect(vim.lsp.buf_get_clients()))
```

Is there something more we should put in here?